user-management/src/main/scala/com/normation/plugins/usermanagement/api/UserManagementApi.scala

```diff
@@ -100,13 +120,117 @@ class UserManagementApiImpl(
 
   implicit val formats = net.liftweb.json.Serialization.formats(NoTypeHints)
 
-  def schemas = UserManagementApi
+  override def schemas = UserManagementApi
 
-  def getLiftEndpoints(): List[LiftApiModule] = {
-    UserManagementApi.endpoints.map(e => e match {
-        case UserManagementApi.GetUserInfo     => GetUserInfo
-        case UserManagementApi.ReloadUsersConf => ReloadUsersConf
-    }).toList
+  def serializeRoleInfo(infos: Map[String, List[String]]): JValue = {
+    infos.map {
+      case(k, v) =>
+        (
+          ("id"    -> k)
+          ~ ("rights" -> v)
+        )
+    }
+  }
+
+  def serializeUser(u: User): JValue = {
+    (
+      ("username"    -> u.username)
+      ~ ("password" -> u.password)
+      ~ ("role"  -> u.role)
+    )
+  }
+
+
+  def serializeRole(rs: Set[Role]): JValue = {
+    val (roles, customs) = rs.partition {
+      case Custom(_) => false
+      case _         => true
+    }
+    (
+      ("role"    -> roles.map(_.name))
+      ~ ("custom" -> customs.flatMap(_.rights.authorizationTypes.map(_.id)).toSeq.sorted)
+    )
+  }
+
+  def extractRoles(req: Req): Option[Set[Role]] = {
+    implicit val formats = DefaultFormats
+    req.json match {
+      case Full(content) =>
+        content match {
+          case JObject(json) =>
+            val roles = json \\ "role" match {
+              case JObject(obj) =>
+                (obj \ "role").extract[Set[String]]
+              case _ => Set.empty
+            }
+            if(roles.isEmpty)
+              None
+            else
+              Some(RoleToRights.parseRole(roles.toSeq).toSet)
+          case _ => None
+        }
+    }
+  }
+
+  def extractAuthzs(req: Req): Option[Set[AuthorizationType]] = {
+    implicit val formats = DefaultFormats
```

Comment:
You should not make your own json extraction, we have methods for this in RestExtractorService or in CompleteJson/OptionnalJson